src/app/containers/BorrowHistory/hooks/useGetBorrowHistoryData.ts

```diff
@@ -0,0 +1,32 @@
+import { useQuery, gql } from '@apollo/client';
+import { useAccount } from 'app/hooks/useAccount';
+
+/** Hook to return data for Borrow History page */
+
+export function useGetBorrowHistoryData() {
+  const account = useAccount();
+  const BORROW_HISTORY = gql`
+    query {
+      user(id: "${account.toLowerCase()}") {
+        borrows {
+          loanId {
+            id
+          }
+          loanToken
+          collateralToken
+          newPrincipal
+          newCollateral
+          interestRate
+          interestDuration
+          collateralToLoanRate
+          timestamp
+          transaction {
+            id
+          }
+        }
+      }
+    }`;
+  const { data, loading } = useQuery(BORROW_HISTORY);
```

Comment:
I think it should go outside of a hook, and maybe it should be not query but selector
And select borrows directly, not as a relation to user:

```suggestion
  const BORROW_HISTORY = gql`
    query getBorrowHistory ($user: ID!) {
        borrows (where: { user: $user } ) {
          loanId {
            id
          }
          loanToken
          collateralToken
          newPrincipal
          newCollateral
          interestRate
          interestDuration
          collateralToLoanRate
          timestamp
          transaction {
            id
          }
        }
    }`;
  const { data, loading } = useQuery(BORROW_HISTORY, { variables: { user: account.toLowerCase() }});
```

Reply:
yes right, i tried to implement it with selecting borrows directly, but with no luck and if i change it to your suggestion i will get an error - Cannot read properties of undefined (reading 'borrows')

Reply:
done